### DISCUSSION TITLE
After turning on MSAA, there is still aliasing, and then turning on post processing. Although anti aliasing is good, why is the color not the original color? I hope to achieve good anti aliasing without turning on post processing. How can I do this

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#6953] [Q&A]
![DMb6VD45zk](https://github.com/google/filament/assets/132528822/36b75132-03ad-4b1b-92fd-9a0c059c3cb4)
After turning on MSAA, there is still aliasing, and then turning on post processing. Although anti aliasing is good, why is the color not the original color? I hope to achieve good anti aliasing without turning on post processing. How can I do this

**Comment by romainguy** [comment]
How many MSAA samples did you use? With post processing on, colors will change a bit because of tone mapping. You can change the tone mapping to "linear" to leave colors intact.

**Comment by Haidiye00** [comment]
How to set linearity

**Reply by poweifeng** [reply]
You have to set the `ColorGrading` param here: https://github.com/google/filament/blob/main/filament/include/filament/View.h#L404

`ColorGrading` has a param call `toneMapper` which you should set to linear: https://github.com/google/filament/blob/main/filament/include/filament/ColorGrading.h#L203

**Comment by Haidiye00** [comment]
ok
thanks
